Replicator/Replicator.cc

```diff
@@ -1284,7 +1289,11 @@ namespace litecore::repl {
     }
 
     void Replicator::delegateCollectionSpecificMessageToWorker(Retained<blip::MessageIn> request) {
-        setMsgHandlerFor3_0_Client(request);
+        // This method is NOT called on the Replicator's queue/thread! It must be thread-safe.
+        if ( !_setMsgHandlerFor3_0_ClientDone ) {
+            enqueue(FUNCTION_TO_QUEUE(Replicator::setMsgHandlerFor3_0_Client), request);
+            waitTillCaughtUp();
```

Comment:
If we have to wait here, could it be already too late, because some messages could have been already handled?

Reply:
No, because this method is being called to dispatch incoming messages. the BLIP connection can't dispatch any more messages until it returns.